noir_stdlib/src/hash/mimc.nr

```diff
@@ -0,0 +1,666 @@
+// mimc-p/p implementation
+// constants are (publicly generated) random numbers, for instance using keccak as a ROM.
+// You must use constants generated for the native field
+// Rounds number should be ~ log(p)/log(exp)
+// For 254 bit primes, exponent 7 and 91 rounds seems to be recommended
+fn mimc<N>(x: Field, k: Field, constants: [Field; N], exp : Field) -> Field {
+    //round 0
+    let mut t = x + k;
+    let mut h = t.pow_32(exp);
+    //next rounds
+    for i in 1 .. constants.len() {
+        t = h + k + constants[i];
+        h = t.pow_32(exp);
+    };
+    h + k
+}
+
+global MIMC_BN254_ROUNDS = 91;
+
+//mimc implementation with hardcoded parameters for BN254 curve.
+fn mimc_bn254<N>(array: [Field; N]) -> Field {
+    //mimc parameters
+    let exponent = 7;
+    //generated from seed "mimc" using keccak256 
+    let constants: [Field; MIMC_BN254_ROUNDS] = [
```

Comment:
I believe we should be able to move these constants out of the func as well. These can be `mimc_bn254_constants` and the others `mimc_feistel_constants` or something similar